File: app/lib/magazzino/models/corridoio_model.dart
// ignore_for_file: non_constant_identifier_names

import 'package:ows_test_v2/generale/models/abstract_model.dart';
import 'package:ows_test_v2/magazzino/models/zona_model.dart';

class CorridoioModel extends AbstractModel {
  int id_corridoio;
  String? cod_corridoio;
  String? desc_corridoio;
  String? cod_alias;
  int? id_magazzino;
  int? id_zona;
  DateTime? data_ins;
  DateTime? data_mod;

  // MagazzinoModel? magazzino; IL MAGAZZINO E' GIA DENTRO A ZONA MODEL QUINDI QUA SAREBBE RIDONDANTE
  ZonaModel? zona;

  CorridoioModel(
      {required this.id_corridoio,
      this.cod_corridoio,
      this.desc_corridoio,
      this.cod_alias,
      this.id_magazzino,
      this.id_zona,
      this.data_ins,
      this.data_mod,
      this.zona});

  @override
  Map<String, dynamic>? toJson() {
    return <String, dynamic>{
      "id_corridoio": id_corridoio,
      "cod_corridoio": cod_corridoio,
      "desc_corridoio": desc_corridoio,
      "cod_alias": cod_alias,
      "id_magazzino": id_magazzino,
      "id_zona": id_zona,
      "data_ins": data_ins,
      "data_mod": data_mod,
      "zona": zona?.toJson(),
    };
  }

  factory CorridoioModel.fromJson(Map<String, dynamic> json) {
    return CorridoioModel(
      id_corridoio: json["id_corridoio"],
      cod_corridoio: json["cod_corridoio"],
      desc_corridoio: json["desc_corridoio"],
      cod_alias: json["cod_alias"],
      id_magazzino: json["id_magazzino"],
      id_zona: json["id_zona"],
      data_ins: DateTime.tryParse(json["data_ins"] ?? ''),
      data_mod: DateTime.tryParse(json["data_mod"] ?? ''),
      zona: json["zona"] != null ? ZonaModel.fromJson(json["zona"]) : null,
    );
  }
}

File: web/Magazzino/Models/CorridoioModel.cs
using System.Data.SqlClient;
using web.Generale;

namespace web.Magazzino.Models
{
   
    public class CorridoioModel : BasicModel
    {
        public int Id_corridoio { get; set; }
        public string? Cod_corridoio { get; set; }
        public string? Desc_corridoio { get; set; }
        public string? Cod_alias { get; set; }
        public int Id_magazzino { get; set; }
        public int Id_zona { get; set; }


        // MagazzinoModel? magazzino; IL MAGAZZINO E' GIA DENTRO A ZONA MODEL QUINDI QUA SAREBBE RIDONDANTE
        public ZonaModel? Zona { get; set; }
        public CorridoioModel()
        {
            NomeTabella = "corridoio";
            GestioneAzienda = false;
            GestioneLogUtenti = false;
        }

       
        public override void FromReader(SqlDataReader reader)
        {
            FromReaderCommon(reader);
            Id_corridoio = GetIntNonNullable(reader, nameof(Id_corridoio));
            Cod_corridoio = GetString(reader, nameof(Cod_corridoio));
            Desc_corridoio = GetString(reader, nameof(Desc_corridoio));
            Cod_alias = GetString(reader, nameof(Cod_alias));
            Id_magazzino = GetIntNonNullable(reader, nameof(Id_magazzino));
            Id_zona = GetIntNonNullable(reader, nameof(Id_zona));

            Zona = new ZonaModel();
            Zona.FromReader(reader);

            
        }
    }
}
